File: joonas/044.cpp
class Solution {
public:
    void reorderList(ListNode* head) {
        vector<ListNode*> v;
        for (ListNode* it = head; it; it = it->next) {
            v.push_back(it);
        }
        int n = v.size(), h = n/2;
        v[h]->next = nullptr;
        for (int i=0; i<h; ++i) {
            v[i]->next = v[n-1-i];
            v[n-1-i]->next = i+1 < h+(n&1) ? v[i+1] : nullptr;
        }
    }
};

File: jlogkim/Season2/044.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    
    public void reorderList(ListNode head) {
         //find final node 
        //and divide the original one with final node make pre finalnode's next pointer null
        ListNode pre=null;
        ListNode node=head;
        
        while(node.next!=null){
            pre=node;
            node=node.next;
        }
        
        if(pre!=null&&pre!=head){
            pre.next=null;
            ListNode next_head=head.next;
            reorderList(next_head);
            node.next=head.next;
            head.next=node;
        }
    }
}